### ISSUE TITLE
Build error, cannot fetch zlib-1.2.11.tar.gz

### ISSUE THREAD

**Issue by dusty-nv** [pytorch/audio#2295]
### 🐛 Describe the bug

```bash
git clone --recursive -b v0.11.0 https://github.com/pytorch/audio torchaudio
cd torchaudio
BUILD_SOX=1 python3 setup.py install
```

This was working fine up until a day or two ago.  Then it started to not be able to download `zlib-1.2.11.tar.gz`.  I don't believe it's a connection issue on my end because I am not experiencing other network issues.

``` bash
-- Git tag: v0.11.0
-- PyTorch dependency: torch
-- Building version 0.11.0+820b383
 --- Initializing submodules
 --- Initialized submodule
 --- Fetching zlib-1.2.11.tar.gz
Traceback (most recent call last):
  File "setup.py", line 181, in <module>
    _main()
  File "setup.py", line 144, in _main
    _fetch_third_party_libraries()
  File "setup.py", line 128, in _fetch_third_party_libraries
    _fetch_archives(_parse_sources())
  File "setup.py", line 122, in _fetch_archives
    torch.hub.download_url_to_file(url, dest, progress=False)
  File "/usr/local/lib/python3.8/dist-packages/torch/hub.py", line 457, in download_url_to_file
    u = urlopen(req)
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

### Versions

Note:  this build is being done in docker, whereas collect_env.py was run outside docker, so not all versions are showing appropriately.  It is with PyTorch 1.11

```
python3 collect_env.py
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (aarch64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Nov 26 2021, 20:14:08)  [GCC 9.3.0] (64-bit runtime)
Python platform: Linux-5.10.65-tegra-aarch64-with-glibc2.29
Is CUDA available: N/A
CUDA runtime version: 11.4.166
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/lib/aarch64-linux-gnu/libcudnn.so.8.3.2
/usr/lib/aarch64-linux-gnu/libcudnn_adv_infer.so.8.3.2
/usr/lib/aarch64-linux-gnu/libcudnn_adv_train.so.8.3.2
/usr/lib/aarch64-linux-gnu/libcudnn_cnn_infer.so.8.3.2
/usr/lib/aarch64-linux-gnu/libcudnn_cnn_train.so.8.3.2
/usr/lib/aarch64-linux-gnu/libcudnn_ops_infer.so.8.3.2
/usr/lib/aarch64-linux-gnu/libcudnn_ops_train.so.8.3.2
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

Versions of relevant libraries:
[pip3] numpy==1.17.4
[conda] Could not collect
```

**Comment by mthrok** [comment]
`1.2.11` has mirror on source forge. https://sourceforge.net/projects/libpng/files/zlib/1.2.11/zlib-1.2.11.tar.gz/download

**Comment by mthrok** [comment]
Reminder: This needs to be fixed for 0.11.1 minor release, otherwise source build would not work.

**Comment by dusty-nv** [comment]
> On `main` branch, can you replace the following lines with the latest?
> 
> https://github.com/pytorch/audio/blob/46ed2b98bc4227f12237f14833380f48c015cf8c/third_party/zlib/CMakeLists.txt#L25-L26
> 
> ```python
> URL https://zlib.net/zlib-1.2.12.tar.gz
> URL_HASH SHA256=91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9
> ```

Thanks @mthrok @nateanl for looking into this today, this worked.

**Comment by coip** [comment]
Could someone chime in on NVIDIA/NeMo#3905 when this is resolved?

**Comment by mthrok** [comment]
Hi @coip - @nateanl has landed #2297, so the build should be working now. We will follow-up on updating zlib to 1.2.12.

**Comment by codeislife99** [comment]
Can this change be backported to v0.11.0 release as well ? If we want to keep the same version we can just make the following change. All of their URLS seem to be in /fossils now. 
```
  URL https://zlib.net/fossils/zlib-1.2.11.tar.gz
  URL_HASH SHA256=c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1
```
